File: android/src/main/java/com/plugin/flutter/zsdk/PrinterUtils.java
package com.plugin.flutter.zsdk;

import com.zebra.sdk.comm.Connection;
import com.zebra.sdk.comm.ConnectionException;
import com.zebra.sdk.printer.ZebraPrinter;
import com.zebra.sdk.printer.ZebraPrinterFactory;
import com.zebra.sdk.printer.ZebraPrinterLanguageUnknownException;

/**
 * Created by luis901101 on 11/9/23.
 */
class PrinterUtils
{
   /**
    * Reboots the printer.
    * Returns true if successfully rebooted, false otherwise
    */
   public static boolean reboot(Connection connection) throws Exception {
      if(!connection.isConnected()) connection.open();
      // Another way would be using the following ZPL command:
      // connection.write("^XA^JUS^XZ".getBytes());
      ZebraPrinter printer = ZebraPrinterFactory.getInstance(connection);
      printer.reset();
      connection.close();
      return true;
   }
}

File: ios/Classes/PrinterUtils.m
//
//  PrinterUtils.m
//  zsdk
//
//  Created by Luis on 11/9/23.
//

#import "PrinterUtils.h"

@implementation PrinterUtils

+ (bool) reboot:(id<ZebraPrinterConnection,NSObject>)connection; {
    NSError *error = nil;
    @try {
        if(![connection isConnected]) [connection open];
//         Another way would be using the following ZPL command:
//        [connection write:[@"^XA^JUS^XZ" dataUsingEncoding:NSUTF8StringEncoding] error:&error];
        
        id<ZebraPrinter,NSObject> printer = [ZebraPrinterFactory getInstance:connection error:&error];
        id<ToolsUtil,NSObject> toolsUtils = [printer getToolsUtil];
        [toolsUtils reset:&error];
        [connection close];
        return true;
    }
    @catch (NSException *e) {
        @throw e;
    }
    return false;
}

@end

